src/Feature/Notification/Feature_Notification.re

```diff
@@ -0,0 +1,315 @@
+open Oni_Core;
+
+module Internal = {
```

Comment:
One thing I was thinking about was that we could remove the `id` generation from being effect-ful to actually being par tof the `model`: 
```
type t = {
   nextId: int,
   notifications: list(notification),
};
```

This keeps the id generation (and message adding) pure. Something I should've done originally!

Not a big deal; as it still encapsulated from the consumer, just something I was thinking about.

Reply:
Definitely. I don't think it matters much in practice though, since it's just used for optimization. I've also considered just using the current time, since that should also be unique, might be nice to have for other purposes as well, and would make this much simpler. I have a note on that, so I'll probably revisit this eventually, but for now I'll just leave it as-is.